### PR TITLE
Add Python 3.11 support

### PR DESCRIPTION
**Pull request by menzenski**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,9 +10,9 @@ keywords = [
 license = "Apache 2.0"
 
 [tool.poetry.dependencies]
-python = "<3.11,>=3.7.1"
+python = "<3.12,>=3.7.1"
 requests = "^2.25.1"
-singer-sdk = "^0.4.5"
+singer-sdk = "^0.13.1"
```
[the remark this answers is not in the document]
Comment (GtheSheep) [reply pyproject.toml:15]: probably may as well while we're here? 😅 afaik it doesn't deprecate any old support so should be fine? happy to test my end on old versions if you want to bump here

also, thanks!

[the remark this answers is not in the document]
Comment (menzenski) [reply pyproject.toml]: Updated!

**Comment by menzenski** [comment]
Updated PR to be "draft" status as I am seeing some issues when trying to use this tap in my meltano project. I will take it out of "draft" status once I resolve those.

**Comment by menzenski** [comment]
Opened https://github.com/meltano/sdk/pull/1365 to add necessary date utility functions to the `sdk` - after that is merged + released I will update this PR to pull that in.

**Review by GtheSheep — APPROVED** [review]

**Pull request merged**